File: app/src/main/java/com/example/collections_and_maps/ui/ViewPagerAdapter.java
package com.example.collections_and_maps.ui;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.collections_and_maps.R;
import com.example.collections_and_maps.ui.benchmark.BenchmarkFragment;

public class ViewPagerAdapter extends FragmentStateAdapter {

    public ViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 0) {
            return BenchmarkFragment.newInstance(R.string.Collections);
        } else if (position == 1) {
            return BenchmarkFragment.newInstance(R.string.Maps);
        } else {
            throw new RuntimeException("Unsupported type");
        }
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/example/collections_and_maps/ui/ViewPagerAdapter.kt
package com.example.collections_and_maps.ui

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.collections_and_maps.R
import com.example.collections_and_maps.ui.benchmark.BenchmarkFragment.Companion.newInstance

class ViewPagerAdapter(activity: FragmentActivity) : FragmentStateAdapter(activity) {

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> newInstance(R.string.Collections)
            1 -> newInstance(R.string.Maps)
            else -> throw RuntimeException("Unsupported type")
        }
    }

    override fun getItemCount(): Int = 2
}
